Contest_API.md

```diff
@@ -621,14 +621,27 @@ The following endpoints are associated with languages:
 
 JSON elements of language objects:
 
-| Name                 | Type            | Required? | Nullable? | Source @WF | Description                                                                                                             |
-| -------------------- | --------------- | --------- | --------- | ---------- | ----------------------------------------------------------------------------------------------------------------------- |
-| id                   | ID              | yes       | no        | CCS        | identifier of the language from table below                                                                             |
-| name                 | string          | yes       | no        | CCS        | name of the language (might not match table below, e.g. if localized)                                                   |
+| Name                 | Type   | Required? | Nullable? | Source @WF | Description                                                           |
+| -------------------- | ------ | --------- | --------- | ---------- | --------------------------------------------------------------------- |
+| id                   | ID     | yes       | no        | CCS        | identifier of the language from table below                           |
+| name                 | string | yes       | no        | CCS        | name of the language (might not match table below, e.g. if localized) |
+| compiler             | Command object | no | yes      | CCS        | Command used for compiling submissions. |
```

Comment:
I get why you used `Command Object` as type here, since you don't want to repeat yourself. However, this is the first time we link to a type that is only defined below (right?). I'm wondering if this is clear enough.

Reply:
Well the list at the top (under "Archive contents") refers to all the types, and they are all (obviously) defined below. That said, up there they are all linked, we should do the same here.

Reply:
A right, sorry. I was indeed referring to the CAF spec. In the API spec we always refer to object definitions below, just not by name. E.g. for the languages endpoint(s) the description is "JSON array of all languages with elements as defined in the table below". If the endpoint table had a type column (and maybe it should), it would say "Language object" and "Array of Language objects" respectively.

In the API spec the definition of the Command objects is literally *directly* below the (very short) table where it is referred to. I think that is close enough for a forward reference to work without further explanation. Also, there is no actual header above the Command object, so we can't actually link to it (although, we could of course change that).

In summary, I suggest no change regarding this. :)

Reply:
OK fair point